src/compile/nodes/DebugTag.ts

```diff
@@ -0,0 +1,44 @@
+import Node from './shared/Node';
+import Tag from './shared/Tag';
+import Block from '../dom/Block';
+import Expression from './shared/Expression';
+import deindent from '../../utils/deindent';
+
+export default class DebugTag extends Node {
+	expression: Expression;
+	
+	constructor(compiler, parent, scope, info) {
+		super(compiler, parent, scope, info);
+		if (info.expression !== null)
+			// Debug when expression nodes change		
+			this.expression = new Expression(compiler, parent, scope, info.expression);
+		else
+			// "Debug all"
+			this.expression = info.expression
+	}
+
+	build(
+		block: Block,
+		parentNode: string,
+		parentNodes: string,
+	) {
+		// Debug all
+		if (this.expression === null) {
+			block.builders.create.addLine('debugger;');
+			block.builders.update.addLine('debugger;');
+		} else {
+			const { dependencies } = this.expression;
+
+			const condition = [...dependencies].map(d => `changed.${d}`).join(' || ');
+
+			const identifiers = [...dependencies].join(', ');
+
+			block.builders.update.addBlock(deindent`
+				if (${condition}) {
+					const { ${identifiers} } = ctx;
+					debugger;
```

Comment:
I think we might need to add a line, to prevent the declarations getting removed by other code transformation tools (e.g. it appears to be happening with Rollup):

```diff
if (${condition}) {
  const { ${identifiers} } = ctx;
+  console.log({ ${identifiers} });
  debugger;
}
```